samples/bluetooth/direct_test_mode/src/transport/dtm_transport.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef DTM_TRANSPORT_H_
+#define DTM_TRANSPORT_H_
+
+#include <zephyr/net/buf.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if (CONFIG_DTM_TRANSPORT_HCI || CONFIG_DTM_REMOTE_HCI_CHILD)
+#define H4_TYPE_CMD 0x01
+#define H4_TYPE_ACL 0x02
+#define H4_TYPE_EVT 0x04
+#define H4_TYPE_ISO 0x05
+#endif
+
+/** @brief DTM transport packet */
+union dtm_tr_packet {
```

Comment:
This approach would not work - the compiler would throw errors when used as a variable.
The risk of misuse is minimal - the union is unpacked in 2 functions per implementation, in both it's assigned to a dedicated variable to avoid mistakes.

Reply:
I understand, it is fine for me.